File: src/app_enums/user_preferences.py
from enum import StrEnum


class UserPreferencesKeys(StrEnum):
    THEME = "theme"
    AUTOMATIC_NOTIFICATIONS = "automatic_notifications"
    LANGUAGE = "language"
    LATEST_GITHUB_VERSION = "latest_github_version"
    DISCLAIMER_SHOWN = "disclaimer_shown"
    LAST_UPDATE_CHECK = "last_update_check"
    DOWNLOAD_DIRECTORY = "download_directory"
    UPGRADE_AVAILABLE = "upgrade_available"

File: lib/enums/user_preferences.dart
enum UserPreferencesKeys {
  theme("theme"),
  automaticNotifications("automatic_notifications"),
  language("language"),
  latestGithubVersion("latest_github_version"),
  disclaimerShown("disclaimer_shown"),
  lastUpdateCheck("last_update_check"),
  downloadDirectory("download_directory"),
  upgradeAvailable("upgrade_available"),
  whatsNewShown("whats_new_shown");

  const UserPreferencesKeys(this.value);
  final String value;
}
